src/Moat.hs

```diff
@@ -122,162 +110,28 @@ data SingSymbol (x :: Symbol)
 instance KnownSymbol x => ToMoatType (SingSymbol x) where
   toMoatType _ = Poly (symbolVal (Proxy @x))
 
--- | A filler type to be used when pretty-printing.
---   The codegen used by shwifty doesn't look at
---   at what a type's type variables are instantiated
---   to, but rather at the type's top-level
---   definition. However,
---   to make GHC happy, you will have to fill in type
---   variables with unused types. To get around this,
---   you could also use something like
---   `-XQuantifiedConstraints`, or existential types,
---   but we leave that to the user to handle.
-type X = Void
```

Comment:
I don't really understand why this is needed and removing it didn't break anything?

Reply:
The documentation here is insufficient. Essentially, when generating code for polymorphic types, GHC will complain unless we fill in all types, since we are using instance lookup. consider

```
data Foo a = Foo a
getMoat ''Foo

prettySwift (@Foo a)
```

You will get a "not in scope: type 'a'" error in the last line. X is used to "fill in" the type at the consumption (pretty-printing) site so GHC doesn't get mad.

Reply:
Maybe this is a remnant of Shwifty? AFAICT this isn't used anywhere?

Reply:
It isn't used anywhere in the test suite because we aren't testing polymorphic codegen (though I believe we should be).
It isn't used anywhere in the Mercury codebase because we don't pretty-print polymorphic models.
Essentially any consumer of the library wanting to generate polymorphic code will have to use _something_ to fill in the types, and 'X' was intended to be that thing. It can safely be removed, but my 2 cents is that the documentation of it should just be improved and its intent made clear there.

Reply:
I believe I have some LHS examples in src/Test.hs.bak which I can turn into golden tests